proposals/text/7462-rag-evaluation.md

```diff
@@ -0,0 +1,225 @@
+- Title: Proposal for presentation of RAG evaluation results
+- Decision driver: David S. Batista
+- Start Date: 2024-04-03
+- Proposal PR: (fill in after opening the PR)
+- Github Issue or Discussion: (only if available, link the original request for this change)
+
+# Summary
+
+Add a new component to Haystack allowing users to evaluate the performance of a RAG model.
+
+# Basic example
+
+```python
+
+data  = {
+    "query_id": ["53c3b3e6, 225f87f7, 8ac473ec, 97d284ca"],
+    "reciprocal_rank": [0.378064, 0.534964, 0.216058, 0.778642],
+    "single_hit": [1, 1, 0, 1],
+    "multi_hit": [0.706125, 0.454976, 0.445512, 0.250522],
+    "context_relevance": [0.805466, 0.410251, 0.750070, 0.361332],
+    "faithfulness": [0.135581, 0.695974, 0.749861, 0.041999],
+    "semantic_answer_similarity": [0.971241, 0.159320, 0.019722, 1],
+    "exact_match": [0, 0, 0, 1]
+}
+
+evaluation = RAGPipelineEvaluation(name="pipeline_1", data=data)
+evaluation.classification_report()
+```
+
+```bash
+{'Reciprocal Rank': 0.448,
+ 'Single Hit': 0.5,
+ 'Multi Hit': 0.540,
+ 'Context Relevance': 0.537,
+ 'Faithfulness': 0.452,
+ 'Semantic Answer Similarity': 0.478,
+ 'Exact Match': 0.442}
+````
+
+# Motivation
+
+RAG models are one of them most popular use cases for Haystack. We are adding support for evaluations metrics, but
+there is no way to present the results of the evaluation.
+
+
+# Detailed design
+
+A new class  `RAGPipelineEvaluation` that receives the results of some or all metric scores over a set of queries/questions
+given to a RAG model.
+
+Example:
+
+```
+data  = {
+    "query_id": ["53c3b3e6, 225f87f7, 8ac473ec, 97d284ca"],
```

Comment:
What's the purpose of this field?

Reply:
To keep track of every specific metric score for each individual query/question as an alternative to store the query/question itself and the ground truth answer and documents.

NOTE: I wrote this proposal thinking of the scenario that we have a dataset of queries/questions which we will use to evalute our pipeline.